12-offer-encoding.md

```diff
@@ -282,6 +373,40 @@ A writer of an offer:
       - MUST set `offer_quantity_max` to 0.
   - otherwise:
     - MUST NOT set `offer_quantity_max`.
+  - If an offer MAY trigger time-spaced invoice requests:
+    - MUST include exactly one of `offer_recurrence_optional` or `offer_recurrence_compulsory`.
+    - MAY use `offer_recurrence_optional` if a payment of a single period would be useful (compatibility with pre-recurrence readers).
+    - MUST NOT use `offer_recurrence_optional` if `offer_recurrence_base` is present.
+  - if it includes either `offer_recurrence_optional` or `offer_recurrence_compulsory`:
+      - MUST set `time_unit` to 0 (seconds), 1 (days), 2 (months) or 3 (years).
+      - MUST set `period` to how often (in `time-unit`) it wants to be paid.
+      - if there is a maximum number of payments:
+        - MUST include `offer_recurrence_limit` with `max_period` set to the maximum number of payments
+        - MUST NOT set `max_period` to 0.
+      - otherwise:
+        - MUST NOT include `offer_recurrence_limit`.
+      - if periods are always at specific time offsets:
+        - MUST include `offer_recurrence_base`
+        - MUST set `basetime` to the initial period time in number of seconds after midnight 1 January 1970, UTC
+        - if `amount` is specified and the node will proportionally reduce the amount charged for a period payed after the start of the period:
+          - MUST set `proportional_amount` to 1 
+        - otherwise:
+          - MUST set `proportional_amount` to 0
+      - otherwise:
+        - MUST NOT include `offer_recurrence_base`.
+      - if payments will be accepted for any time in the current or next period:
+        - SHOULD NOT include `offer_recurrence_paywindow`
+      - otherwise:
+        - MUST include `offer_recurrence_paywindow`
+      - if it includes `offer_recurrence_paywindow`:
+        - MUST set `seconds_before` to the maximum number of seconds prior to a period for which it will accept payment or invoice_request for that period.
+        - MUST set `seconds_after` to the maximum number of seconds into to a period for which it will accept payment or invoice_request for that period.
+        - MAY NOT enforce this for the initial period for offers without `offer_recurrence_base`
+        - SHOULD NOT set `seconds_after` to greater than the maximum number of seconds in a period.
```

Comment:
What if it was set to be longer than the period, would this allow the user to pay for a period’s amount in the next period as well?